bambu_spoolman/broker/automatic_spool_switch.py

```diff
@@ -113,19 +139,48 @@ def _lock_spool(self, tray_id, spool_id):
         save_settings(settings)
         logger.debug("Locked tray {}: {}", tray_id, spool_id)
 
+        # Set active tray in Spoolman
+        # tray_id is calculated as: ams_id * 4 + tray_slot_id (both 0-indexed internally)
+        # So we need to reverse it to get ams_id and tray_slot_id
+        # Both AMS and tray should be 1-indexed for display (1-4)
+        ams_id = (tray_id // 4) + 1
+        tray_slot_id = (tray_id % 4) + 1
+        active_tray_id = f"ams_{ams_id}_tray_{tray_slot_id}"
```

Comment:
I would prefer this to be two separate fields in Spoolman, so you'd have
```
ams: 1
tray: 3
``` as opposed to this "magic" string of `ams_1_tray_3`